### PR TITLE
DOCSP-42794: Laravel Passport

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-42794
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/laravel/DOCSP-42794-passport/user-authentication/#laravel-passport

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/includes/auth/AppServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public function register(): void
+    {
+    }
+
+    /**
+     * Bootstrap any application services.
+     */
+    public function boot(): void
+    {
+        Passport::useAuthCodeModel(AuthCode::class);
```
Comment (GromNaN) [inline docs/includes/auth/AppServiceProvider.php:26]: This example should show that all passport models need to be changed.

**Review by GromNaN — CHANGES_REQUESTED** [review]
A single remark.

**Review by GromNaN — APPROVED** [review]

File: docs/user-authentication.txt [hunk 2]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
    :ref:`laravel-third-party-model` in the Eloquent Model Class guide.
 
+.. _laravel-user-auth-passport:
+
+Laravel Passport
+~~~~~~~~~~~~~~~~
+
+Laravel Passport is an OAuth 2.0 server implementation that offers
+API authentication for Laravel applications. Use Laravel Passport if
+your application requires OAuth2 support.
```
Comment (rustagir) [inline docs/user-authentication.txt:165]: S: maybe you can link to a resource describing Oauth and/or passport?

File: docs/user-authentication.txt [hunk 3]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
+to the ``guards`` array:
+
+.. code-block:: php
+   :emphasize-lines: 7-10
+
+   'guards' => [
+      'web' => [
+         'driver' => 'session',
+         'provider' => 'users',
+      ],
```
Comment (rustagir) [inline docs/user-authentication.txt:220]: Is this other entry default?

File: docs/user-authentication.txt [hunk 4]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
+
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: remove space (will need to adjust emphasized lines)

File: docs/user-authentication.txt [hunk 5]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
+- ``MongoDB\Laravel\Passport\AuthCode``, which extends ``Laravel\Passport\AuthCode``
+- ``MongoDB\Laravel\Passport\Client``, which extends ``Laravel\Passport\Client``
+- ``MongoDB\Laravel\Passport\PersonalAccessClient``, which extends ``Laravel\Passport\PersonalAccessClient``
+- ``MongoDB\Laravel\Passport\RefreshToken``, which extends ``Laravel\Passport\RefreshToken``
+- ``MongoDB\Laravel\Passport\Token``, which extends ``Laravel\Passport\Token``
+
+For example, the following code overrides the default ``Laravel\Passport\AuthCode``
+model class by defining a ``MongoDB\Laravel\Passport\AuthCode`` class and including
+the ``DocumentModel`` trait:
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: possibly clearer explanation:
Something like
```suggestion
The following example code extends the default ``Laravel\Passport\AuthCode``
model class when defining a ``MongoDB\Laravel\Passport\AuthCode`` class and includes
the ``DocumentModel`` trait:
```

File: docs/user-authentication.txt [hunk 6]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
+.. code-block:: php
+
+   class MongoDB\Laravel\Passport\AuthCode extends Laravel\Passport\AuthCode
+   {
+      use MongoDB\Laravel\Eloquent\DocumentModel;
+
+      protected $primaryKey = '_id';
+      protected $keyType = 'string';
+   }
+
```
Comment (rustagir) [inline docs/user-authentication.txt:254]: Q: does the user need to create models for each of the listed classes? If so, that should be more clear in a note or something similar.

File: docs/user-authentication.txt [hunk 7]
```diff
@@ -154,6 +155,107 @@ in the Laravel Sanctum guide.
...
+in the following code:
+
+.. literalinclude:: /includes/auth/AppServiceProvider.php
+   :language: php
+   :emphasize-lines: 26-30
+   :dedent:
+
+You can now use Laravel Passport and MongoDB in your application. For more information, see
+`Laravel Passport <https://laravel.com/docs/{+laravel-docs-version+}/passport>`__ in the
+Laravel documentation.
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: recommend moving the link up to the intro 
```suggestion
The, you can use Laravel Passport with MongoDB in your application.
```

**Review by rustagir — CHANGES_REQUESTED** [review]
a few things!

File: docs/user-authentication.txt (hunk 3 above)
Comment (norareidy) [reply docs/user-authentication.txt:220]: Yes the web entry is added automatically

File: docs/user-authentication.txt (hunk 6 above)
Comment (norareidy) [reply docs/user-authentication.txt:254]: Yeah each one needs a custom model. I changed the wording of the paragraph below (line 246) to clarify that, lmk if it's clear enough

**Review by rustagir — APPROVED** [review]
lgtm!

**Pull request merged**